addons/stock/product.py

```diff
@@ -287,6 +287,30 @@ def _product_available_text(self, cr, uid, ids, field_names=None, arg=False, con
         'orderpoint_ids': fields.one2many('stock.warehouse.orderpoint', 'product_id', 'Minimum Stock Rules'),
     }
 
+    def write(self, cr, uid, ids, vals, context=None):
+        if 'company_id' in vals:
+            move_pool = self.pool.get('stock.move')
+            quant_pool = self.pool.get('stock.quant')
+            for product in self.browse(cr, uid, ids, context=context):
+                if vals['company_id'] == product.company_id.id:
+                    continue
+                active_moves = move_pool.search(
+                    cr, uid, [('product_id', '=', product.id),
+                              ('state', 'not in', ('cancel', 'done')),
```

Comment:
Thinking twice, remove this condition, because although they're not active, if you look for past moves, you'll get an access error.